### PR TITLE
Callback getFolder in LookUpInode 

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
- Call Getfolder in case of LookUpConflicting function.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Added
3. Integration tests - Automated

### CODE REVIEW

File: internal/fs/inode/dir.go [hunk 1]
```diff
@@ -292,6 +292,9 @@ func (d *dirInode) lookUpChildFile(ctx context.Context, name string) (*Core, err
 
 func (d *dirInode) lookUpChildDir(ctx context.Context, name string) (*Core, error) {
 	childName := NewDirName(d.Name(), name)
+	if d.isHNSEnabled && d.bucket.BucketType() == gcs.Hierarchical {
```
Comment (vadlakondaswetha) [inline internal/fs/inode/dir.go:295]: Can you create a helper method for this. i see it is used at multiple places

File: internal/fs/inode/dir.go [hunk 2]
```diff
@@ -292,6 +292,9 @@ func (d *dirInode) lookUpChildFile(ctx context.Context, name string) (*Core, err
...
+		return findExplicitFolder(ctx, d.Bucket(), childName)
+	}
```
Comment (vadlakondaswetha) [inline internal/fs/inode/dir.go:297]: nit: add a new line after this

File: internal/fs/inode/hns_dir_test.go [hunk 3]
```diff
@@ -129,6 +129,27 @@ func (t *HNSDirTest) TestShouldReturnNilWhenGCSFolderNotFoundForInHNS() {
 	assert.Nil(t.T(), result)
 }
 
+func (t *HNSDirTest) TestLookUpChildShouldLookUpConflicting() {
```
Comment (vadlakondaswetha) [inline internal/fs/inode/hns_dir_test.go:132]: name is not clear. what do you mean by lookup conflicting?

Comment (Tulsishah) [reply internal/fs/inode/hns_dir_test.go:132]: The name of the caller function is "lookupConflicting"

Comment (vadlakondaswetha) [reply internal/fs/inode/hns_dir_test.go:132]: I dont see anywhere in the test method the word lookupconflicting.

File: internal/fs/inode/dir.go (hunk 1 above)
Comment (Tulsishah) [reply internal/fs/inode/dir.go]: Done

File: internal/fs/inode/dir.go (hunk 2 above)
Comment (Tulsishah) [reply internal/fs/inode/dir.go:297]: Done

File: internal/fs/inode/hns_dir_test.go (hunk 3 above)
Comment (Tulsishah) [reply internal/fs/inode/hns_dir_test.go]: It was calling function from below if condition.
```
if strings.HasSuffix(name, ConflictingFileNameSuffix) {
	return d.lookUpConflicting(ctx, name)
}
```

Anyways updated test name. Please check

File: internal/fs/inode/dir.go [hunk 4]
```diff
@@ -559,7 +563,7 @@ func (d *dirInode) LookUpChild(ctx context.Context, name string) (*Core, error)
 		return nil, nil
 	case metadata.UnknownType:
 		b.Add(lookUpFile)
-		if d.isHNSEnabled && d.bucket.BucketType() == gcs.Hierarchical {
+		if d.IsBucketHierarchical() {
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/fs/inode/dir.go]: Fixed. Other places too.

File: internal/fs/inode/dir.go [hunk 5]
```diff
@@ -970,3 +974,10 @@ func (d *dirInode) InvalidateKernelListCache() {
 	// Set prevDirListingTimeStamp to Zero time so that cache is invalidated.
 	d.prevDirListingTimeStamp = time.Time{}
 }
+
+func (d *dirInode) IsBucketHierarchical() bool {
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/fs/inode/dir.go]: Done

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**